lighthouse-core/gather/drivers/cri.js

```diff
@@ -16,106 +16,77 @@
  */
 'use strict';
 
-const Driver = require('./driver.js');
-const chromeRemoteInterface = require('chrome-remote-interface');
+const Connection = require('./connection.js');
+const WebSocket = require('ws');
+const http = require('http');
 const port = process.env.PORT || 9222;
 
-const log = require('../../lib/log.js');
-
-class CriDriver extends Driver {
-  constructor() {
-    super();
-
-    /**
-     * Chrome remote interface instance.
-     */
-    this._cri = null;
-  }
-
+class CriConnection extends Connection {
   /**
-   * @return {!Promise<undefined>}
+   * @override
+   * @return {!Promise}
    */
   connect() {
-    return new Promise((resolve, reject) => {
-      if (this._cri) {
-        return resolve();
-      }
-
-      // Make a new tab, stopping Chrome from accidentally giving CRI an "Other" tab.
-      // Also disable the lint check because CRI uses "New" for the function name.
-      /* eslint-disable new-cap */
-      chromeRemoteInterface.New((err, tab) => {
-        if (err) {
-          log.warn('CRI driver', 'cannot create new tab, will reuse tab.', err);
-        }
-
-        chromeRemoteInterface({port: port, chooseTab: tab}, chrome => {
-          this._tab = tab;
-          this._cri = chrome;
-          // The CRI instance is also an EventEmitter, so use directly for event dispatch.
-          this._eventEmitter = chrome;
-          this.enableRuntimeEvents().then(_ => {
-            resolve();
-          });
-        }).on('error', e => reject(e));
+    return this._runJsonCommand('new').then(response => {
+      var url = response.webSocketDebuggerUrl;
+      return new Promise((resolve, reject) => {
+        var ws = new WebSocket(url);
+        ws.on('open', () => {
+          this._ws = ws;
+          resolve();
+        });
+        ws.on('message', data => this.dispatchRawMessage(data));
+        ws.on('close', this.dispose.bind(this));
+        ws.on('error', reject);
       });
-      /* eslint-enable new-cap */
     });
   }
 
-  disconnect() {
+  /**
+   * @return {!Promise<string>}
+   */
+  _runJsonCommand(command) {
     return new Promise((resolve, reject) => {
-      if (!this._tab) {
-        this._cri.close();
-        return resolve();
-      }
-
-      /* eslint-disable new-cap */
-      chromeRemoteInterface.Close({
-        id: this._tab.id
-      }, err => {
-        if (err) {
-          reject(err);
-        } else {
-          resolve();
-        }
+      http.get({
+        hostname: 'localhost',
+        port: port,
+        path: '/json/' + command
+      }, response => {
+        var data = '';
+        response.on('data', chunk => {
+          data += chunk;
+        });
+        response.on('end', _ => {
+          if (response.statusCode === 200) {
+            resolve(JSON.parse(data));
+            return;
+          }
+          reject('Unable to fetch, status: ' + response.statusCode);
```

Comment:
maybe `reject('Unable to fetch webSocketDebuggerUrl, status: ' + response.statusCode);`?